### ISSUE TITLE
Element.animate [ method ] treats percentage values as pixel values

### ISSUE THREAD

**Issue by n-a-m-e** [adobe-webplatform/Snap.svg#383]
https://jsfiddle.net/sjhcnuus/
In the above example I would expect rectangle2 to move above rectangle1 and completely cover it yet all of the attr values fed into rectangle2.animate({}) seem to be converted into pixel based values.

P.S. Snap.svg is awesome keep up the great work

**Comment by triggerdesign** [comment]
+1

**Comment by imfenghuang** [comment]
I have the same problem.

```
    <path id="path" fill="none" stroke="#000000" stroke-miterlimit="10" d="M30.946,424.892c-0.434-5.621,2.645-10.793,3.292-16.357
    c1.178-10.131,0.008-20.927,0.071-31.125c0.123-19.856,4.869-38.831,6.604-58.469c1.642-18.588,5.318-36.955,7.762-55.471
    c1.125-8.523,0.622-17.207,2.018-25.589c2.912-17.479,11.921-36.501,24.746-48.865c2.526-2.435,7.532-7.856,11.553-7.412
    c7.854,0.867,12.568,18.413,14.24,24.808c7.885,30.164,10.969,61.503,15.375,92.221c2.854,19.901,1.394,39.709,3.684,59.516
    c2.453,21.219,8.747,54.02,24.133,70.775c6.369,6.935,14.445,9.882,24.38,7.366c18.243-4.619,25.366-30.661,29.748-45.858
    c6.882-23.871,12.996-47.834,17.788-72.051c4.241-21.436,4.982-43.472,10.044-64.979c3.814-16.203,8.725-32.837,19.054-46.231
    c4.571-5.928,23.446-26.066,33.335-20.751c2.914,1.566,5.604,10.381,7.094,13.29c3.693,7.215,6.559,14.845,9.57,22.361
    c7.654,19.102,14.109,41.139,17.808,61.414c5.122,28.082,7.973,56.437,12.791,84.497c2.797,16.29,4.895,32.839,7.615,48.914
    c1.234,7.289,3.226,17.307,8.074,23.063c7.314,8.685,25.067,4.69,34.143,1.193c25.569-9.854,39.169-28.539,47.61-53.884
    c10.821-32.494,13.985-66.143,23.024-98.991c6.187-22.487,8.944-46.12,18.494-67.734c7.154-16.191,18.018-28.702,36.207-30.953
    c12.735-1.576,34.161-4.147,42.555,8.267c6.076,8.987,7.295,23.56,8.997,34.07c2.146,13.254,3.688,26.518,5.746,39.764
    c2.655,17.097,5.342,34.587,6.247,51.833c1.339,25.516,0.457,50.457,4.529,75.713c2.581,16.008,5.828,41.665,22.469,48.461
    c33.042,13.493,59.92-30.255,67.182-55.778c8.181-28.751,9.903-60.145,14.739-89.326c2.882-17.392,2.908-34.48,5.201-52.014
    c2.052-15.687,2.775-32.409,2.757-48.528c-0.016-13.666,3.38-33.21-1.357-45.907c-8.893-23.833-44.872-27.07-66.171-29.899
    c-34.7-4.609-70.252-4.056-105.428-4.056c-52.437,0-104.874,0-157.311,0c-23.042,0-46.735-1.617-69.71,0.118
    c-23.846,1.801-48.168,6.997-71.782,10.863c-25.993,4.255-53.45,9.966-79.593,9.966"/>

    <text>
        <textPath xlink:href="#path" startOffset="100%" >happy new year</textPath>
    </text>


    <script>
        var svg = Snap("#svg");
        var text= svg.select("text")
        text.select("textPath").animate({
            startOffset:"0%"
        },1000)  
    </script>
```

**Comment by DavidBruchmann** [comment]
@n-a-m-e, The produced svg's code in chrome is like this, means the percent-unit is ignored by snap.js:

`<svg height="100%" version="1.1" width="100%" xmlns="http://www.w3.org/2000/svg">
<desc>Created with Snap</desc>
<defs></defs>
<rect width="50%" height="50%" x="50%" y="50%" fill="#ff0000"></rect>
<rect width="50" height="50" x="50" y="50" fill="#0000ff"></rect>
</svg>`

**Comment by DavidBruchmann** [comment]
@imfenghuang you can make a jfiddle of it too please?

**Comment by imfenghuang** [comment]
@DavidBruchmann this is the demo http://codepen.io/anon/pen/VmJVzE

**Comment by ibrierley** [comment]
Does it say anywhere that a percentage value is valid attribute there ? I didn't think those were valid in svg in this case, but I may be wrong.

**Comment by DavidBruchmann** [comment]
@imfenghuang the %-unit is valid as the first rectangular has it too and it's accepted.

**Comment by DavidBruchmann** [comment]
@imfenghuang thanks for the codepen, can you describe what you expect and what is not working in detail?

**Comment by DavidBruchmann** [comment]
@imfenghuang if you use this snippet
```
<text>
<textPath xlink:href="#path"  startOffset="3000%">happy new year</textPath>
</text>
```
... it seems working right, so the calculation about the startOffset seems being difficult and perhaps even changing if you change the font-size or font-family.

**Comment by DavidBruchmann** [comment]
On  https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/width the description about length-units is this:

> When lengths are used in an SVG attribute, a <length> is instead defined as follows:
> 
> length ::= number ("em" | "ex" | "px" | "in" | "cm" | "mm" | "pt" | "pc" | "%")?
> The unit identifiers in such <length> values must be in lower case.
> 
> Note that the non-property <length> definition also allows a percentage unit identifier. The meaning of a percentage length value depends on the attribute for which the percentage length value has been specified. Two common cases are:

> when a percentage length value represents a percentage of the viewport width or height
> when a percentage length value represents a percentage of the bounding box width or height on a given object.

**Comment by imfenghuang** [comment]
@DavidBruchmann  thx. 

I still have the question.

when i set startOffset="50%" and no animate, the text display in the middle of the the textPath. http://codepen.io/anon/pen/ObKXMd

however, when i use animate, the string "100%" will change, it become the string "100",  the string "%" is gone.  and the display effects of animate is same to use the string “100px”.

**Comment by DavidBruchmann** [comment]
@imfenghuang yeah, that seems then being the same fault of snap.svg like with @n-a-m-e 's svg perhaps.
I don't have an idea if it has some impact to send any pull-request here because the latest commit from @DmitryBaranovskiy was on July 30th. So if anyone want to work on it and a fast result just fork the project, if a PR is accepted here then it won't hurt ;-)

**Comment by DavidBruchmann** [comment]
Hey cool, thats a good message!
Thanks @DmitryBaranovskiy :-)

**Comment by imfenghuang** [comment]
Thanks @DmitryBaranovskiy @DavidBruchmann

**Comment by DavidBruchmann** [comment]
@n-a-m-e this is your code updated with snap.svg-0.4.2dev:
http://codepen.io/dbruchmann/pen/xRvWMW

@imfenghuang your codepen with the new version is this:
http://codepen.io/dbruchmann/pen/woVmxw

... so it seems everything is working fine now, let's know if it's not the case.

**Comment by imfenghuang** [comment]
@DavidBruchmann 
it's working fine
thanks
